Fix: stream end when length of last chunk equal to MAX_CHUNK

PrimozGodec opened the pull request:
This pull request is fixing bug https://github.com/Lukasa/hyper/issues/355

I changed the approach of taking chunks form the generator, such that we know when we reach the last chunk with memorising the next chunk.

PrimozGodec committed:
Fix: no end of the stream when (length of body) % MAX_CHUNK == MAX_CHUNK

Lukasa reviewed (changes requested):
Yup, this looks like it’ll work. Mind adding an entry to the change log and also a test for me?

PrimozGodec commented:
@Lukasa I changed the code and made unit test pass on my machine. I also added a test to test specific change. 
Sorry I can not find a change log. Can you tell me where is it?
Also travis tests are failing it seems that they are failing because of some older code and not because of my changes.

matjazp commented:
@PrimozGodec, CI is fixed now, can you rebase please?

Also, change log is in HISTORY.rst and CONTRIBUTORS.rst.

PrimozGodec commented:
Thank you @matjazp. It is fixed now @Lukasa

Lukasa commented on hyper/http20/stream.py:
```diff
@@ -122,8 +122,17 @@ def file_iterator(fobj):
-            self._send_chunk(chunk, final)
+        # since we need to know when we have a last package we need to know
+        # if there is another package in advance
+        cur_chunk = next(chunks)
```
This will explode if there are no chunks to send: I think you need to write a test case for that and refactor.

PrimozGodec replied:
Done

Lukasa reviewed (approved):
Looks good to me, thanks! :sparkles:
